crates/ruff_cli/src/args.rs

```diff
@@ -76,11 +76,14 @@ pub enum Command {
 pub struct CheckCommand {
     /// List of files or directories to check.
     pub files: Vec<PathBuf>,
-    /// Attempt to automatically fix lint violations.
-    /// Use `--no-fix` to disable.
+    /// Apply automatic fixes to resolve lint violations.
+    /// Use `--no-fix` to disable or `--fix-suggested` to include suggested fixes.
     #[arg(long, overrides_with("no_fix"))]
     fix: bool,
-    #[clap(long, overrides_with("fix"), hide = true)]
+    /// Apply automatic and suggested fixes to resolve lint violations.
+    #[arg(long, overrides_with_all(["fix", "no_fix"]))]
+    fix_suggested: bool,
```

Comment:
What was the reasoning behind using `--fix-suggested` instead of `--unsafe`? Could we also consider `--suggested` or `--with-suggested` or `--include-suggested`? I don't want to derail the PR, but the confusion for me (IIRC) is that it's similar to `--fix` but conceptually independent.

For example, you're now supposed to do `--fix --fix-suggested` (`--fix-suggested` alone as no effect, IIUC), but `--fix-suggested` feels like it should imply `--fix`.

Similarly, you need to do `--diff --fix-suggested` whereas `--diff --fix` would be invalid.

Reply:
Ah gee... I intended for `--fix-suggested` to imply` --fix` but that gets weird with `--diff` which I guess is why Evan went with this approach?

There's lots of discussion elsewhere about why not `--unsafe` but basically the idea was that we wanted consistency between the internal "Suggested" level (which is also going to be displayed in the output eventually) and the user facing concept.

Reply:
My qualms with `--suggested` / `--with-suggested` / `--include-suggested` is that it is not clear that they refer to fixes specifically.

Reply:
Yeah, I went with unsafe originally to expand it to all commands (like --diff --fix-suggested makes less sense than --diff ---unsafe). Maybe --include-unsafe?

Reply:
I think we're going to go with `--unsafe-fixes` and change the internal names somehow (e.g. #7819)